tools/spec-gen-sdk/src/automation/workflowPackage.ts

```diff
@@ -117,13 +117,15 @@ const workflowPkgDetectArtifacts = async (context: WorkflowContext, pkg: Package
  * Copy sdk artifact path to {work_dir}/out/stagedArtifacts
  */
 const workflowPkgSaveSDKArtifact = async (context: WorkflowContext, pkg: PackageData) => {
+  const language = pkg.language ?? getLanguageByRepoName(context.sdkRepoConfig.mainRepository.name);
   const relativeFolderPathParts = pkg.relativeFolderPath.split('/');
-  const serviceName = relativeFolderPathParts[relativeFolderPathParts.indexOf('sdk') + 1];
+  let serviceName = relativeFolderPathParts[relativeFolderPathParts.indexOf('sdk') + 1];
+  if (language.toLocaleLowerCase() === 'go') {
+    serviceName = pkg.relativeFolderPath.replace(/^\/?sdk\//, "");
```

Comment:
Can you double check for both data plane and management plane cases? for management plane sdk, it seems under sdk/resource/{service-name}?

Reply:
I think this is fine
https://github.com/Azure/azure-sdk-for-go/blob/6ed029d78a3101dd906913f517a45ef566566ba1/sdk/azidentity/ci.yml#L34

https://github.com/Azure/azure-sdk-for-go/blob/6ed029d78a3101dd906913f517a45ef566566ba1/sdk/resourcemanager/addons/armaddons/ci.yml#L28

https://github.com/Azure/azure-sdk-for-go/blob/6ed029d78a3101dd906913f517a45ef566566ba1/sdk/security/keyvault/azadmin/ci.yml#L28